### DISCUSSION TITLE
Are failed jobs ever automatically deleted?

### DISCUSSION THREAD

**Discussion by dmurphy-ambra** [jobrunr/jobrunr#526] [Q&A] [answered]
My assumption is that failed jobs are never automatically deleted, since in many cases you would want to re-run them.  However, there are times when a job fails and there is no recovering. In those cases, it would be useful to have such jobs automatically deleted after a period of time.

**Comment by patkujawa-wf** [comment]
I don't believe so, based on my usage (I see a failed job from 2 months ago). I didn't even realize that until you posted!
I don't think I'd want that enabled by default (except maybe with a long duration, e.g. 1 year), but it would be nice to clean the db up automatically even for failures as an opt-in.

**Comment by rdehuyss** [comment, marked as answer]
Indeed, failed jobs are not deleted (as it may be important for auditing proposés).

But, since you guys are running JobRunr Pro, you can use the custom delete policy per job type. So, for you, even failed jobs can be deleted after a configurable duration. More info can be found in the Pro docs.
